.circleci/config.yml

```diff
@@ -36,58 +36,50 @@ parameters:
-    # this is a hot fix for release week
+  check_all:
+    # strictly speaking we don't "need" this(xlarge) but we are continuously adding new examples
     resource_class: xlarge
     executor: ubuntu_vm
     steps:
       - checkout
       - <<: *install_and_start_kurtosis
 
-      # run the Go ethereum partioning test
-      - run: |
-          cd ethereum-network-partition-test
-          go test -timeout 15m  -v
-
-      # run the Starlark ethereum partioning test
-      - run: 
-          name: Ethereum Network Partitioning Starlark Test
+      - run:
+          name: Run the ./scripts/run.sh in each subdirectory
           no_output_timeout: 30m
-          command: kurtosis run ./ethereum-network-partition-test/ --with-subnetworks
-
-      # run the Go Cassandra Network Partitioning Test
-      - run: |
-          cd cassandra-network-partition-test
-          go test -timeout 5m -v
-
-      # run the Starlark Cassandra partioning test
-      - run: |
-          kurtosis run ./cassandra-network-partition-test/ --with-subnetworks
-
-      # run the quickstart Startosis script which should validate the result of the Kurtosis quickstart guide published on: https://docs.kurtosis.com/quickstart/
-      # this can't run with the directory as this uses a sibling package
-      - run: |
-          kurtosis run github.com/kurtosis-tech/awesome-kurtosis/quickstart
-
-      # run the quickstart golang test
-      - run: |
-          cd quickstart/go-test
-          go test -timeout 5m -v
-
-      # run the quickstart ts test
-      - run: |
-          cd quickstart/ts-test
-          npm install -g yarn
-          yarn install
-          yarn build
-          yarn test
+          command: |
+            set -euo pipefail
+            skip_dirs=(".circleci/ .github/ data-package/")
+            for directory in */ ; do
+              if git --no-pager diff --exit-code origin/main...HEAD -- ':.circleci' ":${directory%/}"; then
+                continue
+              fi            
+              if ! echo "$skip_dirs" | grep -q "$directory"; then
+                "./${directory}scripts/run.sh"
+                kurtosis clean -a
+              fi
+            done
+
+  check_all_nightly:
+    # strictly speaking we don't "need" this(xlarge) but we are continuously adding new examples
+    resource_class: xlarge
+    executor: ubuntu_vm
+    steps:
+      - checkout
+      - <<: *install_and_start_kurtosis
 
-      # run the simple-api
-      - run: |
-          ./simple-api/scripts/build.sh
-          kurtosis run ./simple-api/kurtosis-package/
+      - run:
+          name: Run the ./scripts/run.sh in each subdirectory
+          no_output_timeout: 30m
+          command: |
+            set -euo pipefail
+            skip_dirs=(".circleci/ .github/ data-package/")
+            for directory in */ ; do
+              if ! echo "$skip_dirs" | grep -q "$directory"; then
+                "./${directory}scripts/run.sh"
```

Comment:
We used to run the eth package in dry-run mode in the CI, but for real in the nightly run. Not sure if that's something we want to remove. If we want to keep it we can maybe pass an arg to run.sh.
I'm fine with removing it honestly, seems ok for now

Reply:
We do a diff per dir and eth only runs if there are changes. I can add back in the dry run if people complain about the speed!

Reply:
I can create a separate PR after this one that passes dry run as an arg to the non nightly one.